validator/src/main/java/com/google/daq/mqtt/mapping/MappingAgent.java

```diff
@@ -141,25 +144,66 @@ private FamilyDiscoveryConfig getFamilyDiscoveryConfig(String family) {
     return familyDiscoveryConfig;
   }
 
-  private void reconcileDiscovery() {
+  private void mapDiscoveredDevices() {
+    List<Entry<String, Metadata>> mappedDiscoveredEntries = getMappedDiscoveredEntries();
+    Map<String, Metadata> devicesEntriesMap = getDevicesEntries();
+
+    mappedDiscoveredEntries.forEach(entry -> {
+
+      if (devicesEntriesMap.containsKey(entry.getKey())) {
+        System.err.println("Skipping existing device file for family::address = " + entry.getKey());
+        //TODO: update the existing device
+      } else {
+        // UNK denotes unknown device
```

Comment:
I'd extract the "UNK-" to a named class constant.  (And then you could likely remove the comment too since it would be code-as-commend.)